pydocusign/client.py

```diff
@@ -450,12 +450,12 @@ def post_recipient_view(self, authenticationMethod=None,
         data = {
             'authenticationMethod': authenticationMethod,
             'clientUserId': clientUserId,
-            'email': email,
-            'envelopeId': envelopeId,
             'returnUrl': returnUrl,
-            'userId': userId,
-            'userName': userName,
         }
+        for key in ('email', 'userName', 'userId'):
+            # ignore empty values; they won't help with a lookup
+            if locals()[key]:
+                data[key] = locals()[key]
```

Comment:
Using locals() here might seem a bit overkill, but...

Other ways to achive the same result:
- Manually write a dict - but it means one has to write both keys and values and it's a bit ugly (beautiful is better than ugly)(`data.update({"userName": userName, "userId": userId, "email", email})`) et c'est explicite mais moyennement beau 
- Change these 3 values to **kwargs in the function call (as they're not used elsewhere) - but that's breaking the API (is it a public facing API ?).

Using locals is not problematic at all, but it's less "straightfoward code". Thing is, we're using local to help us change a kinda ugly 1 line simple code into a purer 4 line more complex code. Maybe it could be a case of "practicality beats purity"...